upload/catalog/controller/extension/payment/coingate.php

```diff
@@ -90,9 +100,9 @@ public function callback()
 
 
         if (!empty($order_info) && !empty($ext_order) && strcmp($ext_order['token'], $this->request->get['cg_token']) === 0) {
-            $this->setupCoingateClient();
+            $coingateClient = $this->getCoingateClient();
```

Comment:
we send an expired callback after 20 minutes. if the order payment method was changed in that time , we need to check here if payment method is still coingate. or can this case be handled differently?

Reply:
Every time user creates a new order, Opencart creates a new order in the database, and it doesn't change the payment method of the old order. Therefore, I don't think there is a problem.